prov/hostspool/executor.go

```diff
@@ -115,102 +160,153 @@ func (e *defaultExecutor) hostsPoolCreate(originalCtx context.Context, cc *api.C
 		}
 	}
 
-	instances, err := tasks.GetInstances(cc.KV(), taskID, deploymentID, nodeName)
+	instances, err := tasks.GetInstances(cc.KV(), op.taskID, op.deploymentID, op.nodeName)
 	if err != nil {
 		return err
 	}
+
+	return e.allocateHostToInstance(ctx, instances, shareable, filters, op, allocatedResources)
+}
+
+func (e *defaultExecutor) allocateHostToInstance(
```

Comment:
Maybe that this function's name could be _allocateHostsToInstances_ as for each instance the manager trys to allocate a host ?

Reply:
ok will change